openpype/plugins/publish/extract_review.py

```diff
@@ -996,70 +993,119 @@ def get_letterbox_filters(
-                ).format(ratio, fill_color_hex, fill_color_alpha)
+                    "drawbox=0:0:{widht}:round("
+                    "({height}-({widht}*(1/{ratio})))/2)"
+                    ":t=fill:c={color}@{alpha}"
+                ).format(
+                    widht=output_width,
+                    height=output_height,
+                    ratio=ratio,
+                    color=fill_color_hex,
+                    alpha=fill_color_alpha
+                )
 
                 bottom_box = (
-                    "drawbox=0:ih-round((ih-(iw*(1/{0})))/2)"
-                    ":iw:round((ih-(iw*(1/{0})))/2):t=fill:c={1}@{2}"
-                ).format(ratio, fill_color_hex, fill_color_alpha)
+                    "drawbox=0:{height}-round("
+                    "({height}-({widht}*(1/{ratio})))/2)"
+                    ":{widht}:round(({height}-({widht}"
+                    "*(1/{ratio})))/2):t=fill:"
+                    "c={color}@{alpha}"
+                ).format(
+                    widht=output_width,
+                    height=output_height,
+                    ratio=ratio,
+                    color=fill_color_hex,
+                    alpha=fill_color_alpha
+                )
 
-                output.extend([top_box, bottom_box])
+                if height_letterbox > 0:
+                    output.extend([top_box, bottom_box])
 
             if line_color_alpha > 0 and line_thickness > 0:
                 top_line = (
-                    "drawbox=0:round((ih-(iw*(1/{0})))/2)-{1}:iw:{1}:"
-                    "t=fill:c={2}@{3}"
+                    "drawbox=0:round(({height}-({widht}"
+                    "*(1/{ratio})))/2)-{l_thick}:{widht}:{l_thick}:"
+                    "t=fill:c={l_color}@{l_alpha}"
                 ).format(
-                    ratio, line_thickness, line_color_hex, line_color_alpha
+                    widht=output_width,
+                    height=output_height,
+                    ratio=ratio,
+                    l_thick=line_thickness,
+                    l_color=line_color_hex,
+                    l_alpha=line_color_alpha
                 )
                 bottom_line = (
-                    "drawbox=0:ih-round((ih-(iw*(1/{})))/2)"
-                    ":iw:{}:t=fill:c={}@{}"
+                    "drawbox=0:{height}-round(({height}-({widht}"
+                    "*(1/{ratio})))/2)"
+                    ":{widht}:{l_thick}:t=fill:c={l_color}@{l_alpha}"
                 ).format(
-                    ratio, line_thickness, line_color_hex, line_color_alpha
+                    widht=output_width,
+                    height=output_height,
+                    ratio=ratio,
+                    l_thick=line_thickness,
+                    l_color=line_color_hex,
+                    l_alpha=line_color_alpha
                 )
-                output.extend([top_line, bottom_line])
+                if height_letterbox > 0:
```

Comment:
All these conditions should not be here or should be at the beginning because this whole function can be skipped when it's true.

Reply:
eh.. yeah.. good catch ;)